### ISSUE TITLE
CI: test failure PrimaryAllocationIT.testForceStaleReplicaToBePromotedToPrimaryOnWrongNode

### ISSUE THREAD

**Issue by atorok** [elastic/elasticsearch#37345]
## Example build failure
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/1198/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=ubuntu&&virtual/174/console

## Reproduction line 
does not reproduce locally
```
/gradlew :server:integTest -Dtests.seed=ED29DBC5949B19E9 -Dtests.class=org.elasticsearch.cluster.routing.PrimaryAllocationIT -Dtests.method="testForceStaleReplicaToBePromotedToPrimaryOnWrongNode" -Dtests.security.manager=true -Dtests.locale=cs -Dtests.timezone=Europe/Kaliningrad -Dcompiler.java=11 -Druntime.java=8
```

## Example relevant log:
```
10:28:30 ERROR   4.30s J3 | PrimaryAllocationIT.testForceStaleReplicaToBePromotedToPrimaryOnWrongNode <<< FAILURES!
10:28:30    > Throwable #1: java.lang.NullPointerException
10:28:30    > 	at __randomizedtesting.SeedInfo.seed([ED29DBC5949B19E9:D46FFB23117F3451]:0)
10:28:30    > 	at org.elasticsearch.cluster.routing.PrimaryAllocationIT.testForceStaleReplicaToBePromotedToPrimaryOnWrongNode(PrimaryAllocationIT.java:282)
10:28:30    > 	at java.lang.Thread.run(Thread.java:748)
```

## Frequency
up to 8-9 a day.

Possibly related to: #35497

**Comment by atorok** [comment]
@original-brownbear could it be related to #37226 ?

**Comment by atorok** [comment]
Muted in 8e5e38574e9

**Comment by original-brownbear** [comment]
@atorok yea that def. seems related, I should have a fix in a few minutes :)

**Comment by matriv** [comment]
New failures: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=zulu11,nodes=virtual&&linux/210/console

Reproduce:
```
./gradlew :server:integTest \
  -Dtests.seed=E77709B66B5E760A \
  -Dtests.class=org.elasticsearch.cluster.routing.PrimaryAllocationIT \
  -Dtests.method="testForceStaleReplicaToBePromotedToPrimaryOnWrongNode" \
  -Dtests.security.manager=true \
  -Dtests.locale=es-CL \
  -Dtests.timezone=Europe/Brussels \
  -Dcompiler.java=11 \
  -Druntime.java=11
```

Couldn't reproduce locally.

```
ERROR   1128s J4 | PrimaryAllocationIT.testForceStaleReplicaToBePromotedToPrimaryOnWrongNode <<< FAILURES!
  2> 	at java.base@11.0.2/java.lang.Thread.run(Thread.java:834)
  2> "elasticsearch[node_td4][[timer]]" ID=3401 TIMED_WAITING
```

**Comment by matriv** [comment]
And another one: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java8,nodes=virtual&&linux/210/console

```
ERROR   16.3s J4 | PrimaryAllocationIT.testForceStaleReplicaToBePromotedToPrimaryOnWrongNode <<< FAILURES!
   > Throwable #1: java.lang.NullPointerException
   > 	at __randomizedtesting.SeedInfo.seed([EF58BD1E4C92C848:D61E9DF8C976E5F0]:0)
   > 	at org.elasticsearch.cluster.routing.PrimaryAllocationIT.testForceStaleReplicaToBePromotedToPrimaryOnWrongNode(PrimaryAllocationIT.java:313)
   > 	at java.lang.Thread.run(Thread.java:748)
  1> [2019-01-29T11:27:02,923][INFO ][o.e.c.r.PrimaryAllocationIT] [testForceStaleReplicaToBePromotedToPrimary] before test
  1> [2019-01-29T11:27:02,923][INFO ][o.e.c.r.PrimaryAllocationIT] [testForceStaleReplicaToBePromotedToPrimary] [PrimaryAllocationIT#testForceStaleReplicaToBePromotedToPrimary]: setting up test
  1> [2019-01-29T11:27:02,924][INFO ][o.e.t.InternalTestCluster] [testForceStaleReplicaToBePromotedToPrimary] Setup InternalTestCluster [TEST-CHILD_VM=[4]-CLUSTER_SEED=[-4661995235852411665]-HASH=[1419D940F93]-cluster] with seed [BF4D440886741CEF] using [0] dedicated masters, [0] (data) nodes and [0] coord only nodes (min_master_nodes are [auto-managed])
  1> [2019-01-29T11:27:02,924][INFO ][o.e.c.r.PrimaryAllocationIT] [testForceStaleReplicaToBePromotedToPrimary] [PrimaryAllocationIT#testForceStaleReplicaToBePromotedToPrimary]: all set up test
  1> [2019-01-29T11:27:02,924][INFO ][o.e.c.r.PrimaryAllocationIT] [testForceStaleReplicaToBePromotedToPrimary] --> starting 3 nodes, 1 master, 2 data
  1> [2019-01-29T11:27:02,928][INFO ][o.e.e.NodeEnvironment    ] [testForceStaleReplicaToBePromotedToPrimary] using [1] data paths, mounts [[/ (/dev/sda1)]], net usable_space [465.7gb], net total_space [503.9gb], types [ext4]
  1> [2019-01-29T11:27:02,928][INFO ][o.e.e.NodeEnvironment    ] [testForceStaleReplicaToBePromotedToPrimary] heap size [491mb], compressed ordinary object pointers [true]
  1> [2019-01-29T11:27:02,929][INFO ][o.e.n.Node               ] [testForceStaleReplicaToBePromotedToPrimary] node name [node_tm0], node ID [uGBxeepJTWy3MSBE9paOtQ]
  1> [2019-01-29T11:27:02,929][INFO ][o.e.n.Node               ] [testForceStaleReplicaToBePromotedToPrimary] version[6.7.0-SNAPSHOT], pid[14028], build[unknown/unknown/Unknown/Unknown], OS[Linux/4.4.0-141-generic/amd64], JVM[Oracle Corporation/Java HotSpot(TM) 64-Bit Server VM/1.8.0_202/25.202-b08]
```

**Comment by original-brownbear** [comment]
@matriv I'm on it, sorry for the noise.

**Comment by matriv** [comment]
@original-brownbear No worries :-)

**Comment by original-brownbear** [comment]
The NPE when calling `org.elasticsearch.test.InternalTestCluster#getNodeNames` is quite interesting, the only way I can see this happening is via some concurrent modification of `org.elasticsearch.test.InternalTestCluster#nodes` that is happening in `6.x` but not in `master`.
Going through the differences between the two now to see where this could be coming from.

**Comment by original-brownbear** [comment]
This will be resolved by #39168
